pkg/transform/oauth/oauth.go

```diff
@@ -171,3 +173,57 @@ func (oauth *CRD) GenYAML() ([]byte, error) {
 
 	return yamlBytes, nil
 }
+
+// Validate validate oauth providers
+func Validate(identityProviders []IdentityProvider) error {
+	var err error
+	serializer := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for _, identityProvider := range identityProviders {
+		switch identityProvider.Kind {
+		case "BasicAuthPasswordIdentityProvider":
+			err = validateBasicAuthProvider(serializer, identityProvider)
+		case "GitHubIdentityProvider":
+			err = validateGithubProvider(serializer, identityProvider)
+		case "GitLabIdentityProvider":
+			err = validateGitLabProvider(serializer, identityProvider)
+		case "GoogleIdentityProvider":
+			err = validateGoogleProvider(serializer, identityProvider)
+		case "HTPasswdPasswordIdentityProvider":
+			err = validateHTPasswdProvider(serializer, identityProvider)
+		case "KeystonePasswordIdentityProvider":
+			err = validateKeystoneProvider(serializer, identityProvider)
+		case "LDAPPasswordIdentityProvider":
+			err = validateLDAPProvider(serializer, identityProvider)
+		case "OpenIDIdentityProvider":
+			err = validateOpenIDProvider(serializer, identityProvider)
+		case "RequestHeaderIdentityProvider":
+			err = validateRequestHeaderProvider(serializer, identityProvider)
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
Do you want to stop processing if there is an error in ANY of the providers?  If so, okay. If not, the for loop definitely will exit the first time one of the providers fails to validate the rest will be skipped.